src/routes/view-invoice.tsx

```diff
@@ -0,0 +1,143 @@
+import { Alert, AlertTitle, Box } from '@mui/material';
+import { visuallyHidden } from '@mui/utils';
+import {
+  useParams,
+  useSearchParams,
+  useNavigate,
+  Link,
+} from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { useGetInvoice, useUpdateInvoice } from '../hooks/invoices.hooks.ts';
+import { z } from 'zod';
+import { InvoiceSchema } from '../types/Invoice.ts';
+
+import { InvoiceForm, InvoiceFormProps } from '../components/InvoiceForm.tsx';
+import Button from '@mui/material/Button';
+import Icon from '@mui/material/Icon';
+
+export const ViewInvoicePage = () => {
+  const { t } = useTranslation();
+  const { id } = z.object({ id: z.string() }).parse(useParams());
+  const { data: invoice } = useGetInvoice(id);
+  const navigate = useNavigate();
+
+  const [searchParams] = useSearchParams();
+
+  const isEditMode = searchParams.get('mode') === 'edit';
+
+  const updateInvoiceMutation = useUpdateInvoice();
+
+  const onSubmit = async (data: any) => {
+    if (!isEditMode || !id) return;
+
+    await updateInvoiceMutation.mutateAsync(
```

Comment:
mutateAsync may throw an error. When you are using `onSuccess` and `onError`, simple `mutate` is rather more suitable.